### PR TITLE
Span attributes

### LINKED ISSUE #11: Cleaner spans
This is a meta issue about a few things that could be improved about metadata of the spans. I'm doing some tests about what would be best, and should push a PR at some point.

- The root span is named "test run" with no meaningful attributes. It would be nice to be able to customize it. Not sure how though, maybe an env variable for start?
- The span name uses `node.nodeid`  for its name instead of `node.name` which doesn't include the filename. This info can be found in `code.filepath` anyway.
- `code.function` should use `item.name`.
- `code.lineno` can be 0 when running from a Tavern yaml file, maybe only include it when `> 0`. Otel supports ints so no need to `str(line_number)`.
- (wont do) `start_as_current_span` has optional attributes that can auto record exceptions https://opentelemetry-python.readthedocs.io/en/latest/api/trace.html#opentelemetry.trace.Tracer.start_as_current_span

### PR DESCRIPTION
**Pull request by Yamakaky**
Sorry for the delay.

I created https://github.com/Yamakaky/tests-tavern to have a test run with both tavern and pytest, maybe it can be added in your tests. For each commit here, I'll add a collector.log so that we can see the attributes generated.

Fixes #11

Implemented:
- [X] Configurable root span name via env variable
  - [ ] maybe other means, different env variable ?
- [X] Use node.id in an attribute and node.name in span name
- [X] Set code.function to item.name
- [X] Optional code.lineno
- [ ] record tavern test parameters as attributes
- [x] Create attributes to differentiate main span and test span
- [x] Set test run status

### CODE REVIEW

File: src/pytest_opentelemetry/instrumentation.py [hunk 1]
```diff
@@ -96,13 +107,15 @@ def pytest_exception_interact(
             )
         )
 
-    @staticmethod
-    def pytest_runtest_logreport(report: TestReport) -> None:
+    def pytest_runtest_logreport(self, report: TestReport) -> None:
         if report.when != 'call':
+            trace.get_current_span().set_attribute(f"when_{report.when}", report.outcome)
```
Comment (chrisguidry) [inline src/pytest_opentelemetry/instrumentation.py:112]:
```suggestion
            trace.get_current_span().set_attribute(f"pytest.when_{report.when}", report.outcome)
```
Maybe?

**Review by chrisguidry — COMMENTED** [review]
I like it!  I think there are some test failures, let me know if you need a hand with them

File: src/pytest_opentelemetry/instrumentation.py (hunk 1 above)
Comment (Yamakaky) [reply src/pytest_opentelemetry/instrumentation.py:112]: Oh no, I just forgot to remove that XD

**Comment by Yamakaky** [comment]
Should be good now. I think we should merge that then have the test parameters part in a separate PR since I don't know how to do it for now. Just validate that the env variable name is OK with you?

**Review by chrisguidry — APPROVED** [review]
Looks great!  I agree that we should merge this now to keep the changes small.  I'll cut a new release in a moment...

**Pull request merged**